invisible_cities/reco/deconv_functions_test.py

```diff
@@ -0,0 +1,174 @@
+import os
+import string
+import random
+import numpy  as np
+import pandas as pd
+
+
+from hypothesis              import given
+from hypothesis              import reproduce_failure
+from hypothesis.strategies   import text
+from hypothesis.strategies   import floats
+from hypothesis.strategies   import tuples
+from hypothesis.strategies   import composite
+from hypothesis.strategies   import lists
+from hypothesis.extra.numpy  import arrays
+
+from .. reco.deconv_functions import cut_and_redistribute_df
+from .. reco.deconv_functions import drop_isolated_sensors
+from .. reco.deconv_functions import interpolateSignal
+from .. reco.deconv_functions import deconvolutionInput
+from .. reco.deconv_functions import deconvolve
+from .. reco.deconv_functions import richardson_lucy
+
+from .. core.core_functions   import in_range
+from .. core.core_functions   import shift_to_bin_centers
+from ..   io.dst_io           import load_dst
+
+from scipy.stats             import multivariate_normal
+
+characters = string.ascii_letters
+
+@composite
+def dst(draw, dimension=[3,10]):
+    columns = draw(lists(text(characters, min_size=5), min_size=dimension[0], max_size=dimension[1], unique=True))
+    values  = draw(lists(arrays(float, 100, floats(1, 1e3, allow_nan=False, allow_infinity=False)), min_size=len(columns), max_size=len(columns)))
+
+    return pd.DataFrame({k:v for k, v in zip(columns, values)})
+
+@given(dst())
+def test_cut_and_redistribute_df(df):
+    cut_var       = random.choice (df.columns)
+    redist_var    = random.choices(df.columns, k=3)
+    cut_condition = f'{cut_var} > {df[cut_var].mean()}'
+    cut_function  = cut_and_redistribute_df(cut_condition, redist_var)
+    df_cut        = cut_function(df)
+    df_cut_manual = df[df[cut_var] > df[cut_var].mean()]
+
+    assert len(df_cut) == len(df_cut_manual)
+
+    if (len(df_cut)) != 0:
+        for c in df_cut.columns:
+            if c in redist_var:
+                assert np.allclose(df_cut[c].values, df_cut_manual[c].values * df[c].values.sum() / df_cut_manual[c].values.sum())
+            else:
+                assert np.allclose(df_cut[c].values, df_cut_manual[c].values)
+
+def test_drop_isolated_sensors():
+    size          = 20
+    dist          = [10., 10.]
+    x, y          = random.choices(np.linspace(-200, 200, 41), k=size), random.choices(np.linspace(-200, 200, 41), k=size)
+    q             = np.random.uniform(0,  20, size)
+    e             = np.random.uniform(0, 200, size)
+    df            = pd.DataFrame({'X':x, 'Y':y, 'Q':q, 'E':e})
+    drop_function = drop_isolated_sensors(dist, ['E'])
+    df_cut        = drop_function(df)
+
+    if len(df_cut > 0):
+        assert np.isclose(df_cut.E.sum(), df.E.sum())
+
+    for row in df_cut.itertuples(index=False):
+        n_neighbours = len(df_cut[in_range(df_cut.X, row.X - dist[0], row.X + dist[0] + 1e-8) & in_range(df_cut.Y, row.Y - dist[1], row.Y + dist[1] + 1e-8)])
+        assert n_neighbours > 1
```

Comment:
I don't like that `+1e-8`. I think this is a needed feature in `in_range`: add arguments to decide which comparison to use on each side. For now maybe you can increase the distance to `10.1`?

Reply:
Also split the line in two